students/jesse_miller/session09/donor_models.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python3
+'''
+Let's try this by following the directions.  So, this is going to be where the
+donor manipulation occurs.  The goal is to compartmentalize everything so that
+functions can be called by other programs if and when needed.
+'''
+
+
+class Donor:
+    '''
+    This will be slightly easier in a sense because we're only dealing with a
+    single donor.  However, I get to look forward to pairing everything down to
+    an interesting, if not insane, degree from previous attempts.
+    '''
+
+    def __init__(self, name):
+        self.name = name
+        self.donations = [] #?  We'll try it.
+        self.count = 0
+        self.total = 0
+        self.average = 0
+
+
+    def donation_add(self, new_donation):
+        '''
+        This should append donations to the list
+        '''
+        self.donations.append(new_donation)
+
+    @property
+    def donation_count(self):
+        '''
+        Fairly obvious, counts the number of donations
+        '''
+        return len(self.donations)
+
+
+    @donation_count.setter
+    def donation_count(self, value):
+        self.count = value
+
+
+    @property
+    def donation_total(self):
+        '''
+        Again, obvious, adds the donations up
+        '''
+        return sum(self.donations)
+
+
+    @donation_total.setter
+    def donation_total(self, value):
+        self.total = value
+
+
+    @property
+    def donation_average(self):
+        '''
+        Averages the donations
+        '''
+        return self.donation_total / self.donation_count
+
+
+    @donation_average.setter
+    def donation_average(self, value):
+        self.average = value
+
+
+################################################################################
+'''
+End Donors
+'''
+################################################################################
+
+class DonorCollection:
+    '''
+    This class will handle the donors dictionary, as well as the collections.
+    Some of the things we can look forward to:
+    Create a donor
+    Find a donor
+    List donors
+    Donor reports
+    Data saving and reloading.
+    '''
+    def __init__(self):
+        '''
+        Initializing and building the default dict
+        '''
+        self.donors_dict = {'Robert Smith': [435.56, 125.23, 357.10],
```

Comment:
_self.donors_dict_ should instead have the key (the name of the donor is fine) point to a _Donor_ object, instead of pointing to a list of donations. This will make it easier to add new donor _features_ by simply modifying the _Donor_ class.

Reply:
So:
donor = "firstname lastname"

donor.donors_dict (functions)

Am I understanding this right?

Reply:
Well, 

total = Donors(donor.donations_total) 

for example?

Reply:
Not exactly.

Let's simplify the initialization, just do:

```
self.donors_dict = {}
```
Then, when you add a new donor, the process will go like this:

1. Ask the user for the name & last name of the new donor.
2. Check if that donor already exists in _donors_dict_.
3. Call _donor_creation_, which will create a new _Donor_ object and add it to _donors_dict_ inside of the _DonorCollection_ object.

Reply:
Right now, the data in the dictionary is not consistent, since the example you're loading has each key pointing to a list of integers, not to a Donor object.